src/backend/distributed/commands/multi_copy.c

```diff
@@ -246,15 +246,16 @@ static CopyConnectionState * GetConnectionState(HTAB *connectionStateHash,
 static CopyShardState * GetShardState(uint64 shardId, HTAB *shardStateHash,
 									  HTAB *connectionStateHash, bool stopOnFailure,
 									  bool *found, bool shouldUseLocalCopy, CopyOutState
```

Comment:
This function could use some refactoring in the future to reduce its large number of parameters.

Reply:
Makes sense. We have many functions like this, we should probably make use of structs a bit more.